deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/optimize/listeners/TimeIterationListener.java

```diff
@@ -23,36 +23,48 @@
 import lombok.extern.slf4j.Slf4j;
 import org.deeplearning4j.nn.api.Model;
 import org.deeplearning4j.optimize.api.BaseTrainingListener;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import java.io.Serializable;
 import java.util.Date;
 import java.util.concurrent.atomic.AtomicLong;
 
 @Slf4j
 public class TimeIterationListener extends BaseTrainingListener implements Serializable {
-    private long start;
-    private int iterationCount;
-    private AtomicLong iterationCounter = new AtomicLong(0);
+
+    private final int frequency;
+    private final long start;
+    private final int iterationCount;
+    private final AtomicLong iterationCounter = new AtomicLong(0);
 
     /**
      * Constructor
      * @param iterationCount The global number of iteration for training (all epochs)
      */
     public TimeIterationListener(int iterationCount) {
+        this(1, iterationCount);
+    }
+
+    /**
+     * Constructor
+     * @param frequency frequency with which to print the remaining time
+     * @param iterationCount The global number of iteration for training (all epochs)
+     */
+    public TimeIterationListener(int frequency, int iterationCount) {
+        this.frequency = frequency;
         this.iterationCount = iterationCount;
-        start = System.currentTimeMillis();
+        this.start = System.currentTimeMillis();
     }
 
     @Override
     public void iterationDone(Model model, int iteration, int epoch) {
         long currentIteration = iterationCounter.incrementAndGet();
-        long elapsed = System.currentTimeMillis() - start;
-        long remaining = (iterationCount - currentIteration) * elapsed / currentIteration;
-        long minutes = remaining / (1000 * 60);
-        Date date = new Date(start + elapsed + remaining);
-        log.info("Remaining time : " + minutes + "mn - End expected : " + date.toString());
+        if (iteration % frequency == 0) {
```

Comment:
Could you define a default frequency so we don't break old behavior?

Reply:
I did, the old constructor calls the new one with a frequency of 1. Then it is the same as before.